### PR TITLE
feature(zjow): add dmc2gym env

### PR DESCRIPTION
**Pull request by zjowowen**
Add dmc2gym envs.
Add Dockerfile and workflow for generating DI-engine image with dm_control and dmc2gym.

### CODE REVIEW

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+def dmc2gym_state_space(dim, minimum=-np.inf, maximum=np.inf, dytpe=np.float32) -> Box:
+    return Box(np.repeat(minimum, dim).astype(dytpe), np.repeat(maximum, dim).astype(dytpe), dtype=dytpe)
+
+
+def dmc2gym_action_space(dim, minimum=-1, maximum=1, dytpe=np.float32) -> Box:
+    return Box(np.repeat(minimum, dim).astype(dytpe), np.repeat(maximum, dim).astype(dytpe), dtype=dytpe)
+
+
+def dmc2gym_reward_space(minimum=0, maximum=1) -> tuple:
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:23]: reward space should also be `gym.spaces`

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+}
+
+
+@ENV_REGISTRY.register('dmc2gym')
+class DMC2GymEnv(BaseEnv):
+
+    def __init__(self, cfg: dict = {}) -> None:
+        assert cfg.domain_name in dmc2gym_env_info
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:92]: `assert cfg.domain_name in dmc2gym_env_info, '{}/{}'.format(cfg.domain_name, dmc2gym_env_info.keys())`

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        assert cfg.task_name in dmc2gym_env_info[cfg.domain_name]
+        self._domain_name = cfg.domain_name
+        self._task_name = cfg.task_name
+
+        self._cfg = cfg
+        self._init_flag = False
+
+        #to do
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:100]: # TODO(xxx) something

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            if self._replay_path is not None:
+                pass
+            self._init_flag = True
+
+        if hasattr(self, '_seed') and hasattr(self, '_dynamic_seed') and self._dynamic_seed:
+            np_seed = 100 * np.random.randint(1, 1000)
+            self._env.seed(self._seed + np_seed)
+        elif hasattr(self, '_seed'):
+            self._env.seed(self._seed)
+        #self._observation_space = self._env.observation_space
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:120]: remove this line

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        if self._init_flag:
+            self._env.close()
+        self._init_flag = False
+
+    def seed(self, seed: int, dynamic_seed: bool = True) -> None:
+        self._seed = seed
+        self._dynamic_seed = dynamic_seed
+        np.random.seed(self._seed)
+
+    def step(self, action: Union[int, np.ndarray]) -> BaseEnvTimestep:
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:136]: why `int` here

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+from gym.spaces import Box
+import numpy as np
+from ding.envs import BaseEnv, BaseEnvTimestep
+from ding.torch_utils import to_ndarray
+from ding.utils import ENV_REGISTRY
+import dmc2gym
+
+
+def dmc2gym_observation_space(dim, minimum=-np.inf, maximum=np.inf, dytpe=np.float32) -> Box:
+    return Box(np.repeat(minimum, dim).astype(dytpe), np.repeat(maximum, dim).astype(dytpe), dtype=dytpe)
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:12]: dtype

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
+from typing import Union, Optional
+import gym
+from gym.spaces import Box
+import numpy as np
+from ding.envs import BaseEnv, BaseEnvTimestep
+from ding.torch_utils import to_ndarray
+from ding.utils import ENV_REGISTRY
+import dmc2gym
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:8]: add it into `setup.py`

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        #to do
+        self._replay_path = None
+
+        self._observation_space = dmc2gym_env_info[cfg.domain_name][cfg.task_name]["observation_space"]
+        self._action_space = dmc2gym_env_info[cfg.domain_name][cfg.task_name]["action_space"]
+        self._reward_space = dmc2gym_env_info[cfg.domain_name][cfg.task_name]["reward_space"]
+
+    def reset(self) -> np.ndarray:
+        if not self._init_flag:
+            self._env = dmc2gym.make(domain_name=self._domain_name, task_name=self._task_name)
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:109]: you should add configuration like [this](https://github.com/MishaLaskin/curl/blob/master/train.py#L158)

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
+from typing import Union, Optional
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:1]: add envtest for this file

File: setup.py [hunk 10]
```diff
@@ -53,7 +53,6 @@
         'torch>=1.1.0,<=1.10.0',  # PyTorch 1.10.0 is available, if some errors, you need to do something like https://github.com/opendilab/DI-engine/discussions/81
         'pyyaml<6.0',
         'easydict==1.9',
-        'protobuf<=3.20.1',
```
Comment (PaParaZz1) [inline setup.py:56]: why remove this

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+class DMC2GymEnv(BaseEnv):
+
+    def __init__(self, cfg: dict = {}) -> None:
+        assert cfg.domain_name in dmc2gym_env_info, '{}/{}'.format(cfg.domain_name, dmc2gym_env_info.keys())
+        assert cfg.task_name in dmc2gym_env_info[
+            cfg.domain_name], '{}/{}'.format(cfg.task_name, dmc2gym_env_info[cfg.domain_name].keys())
+
+        self._domain_name = cfg.domain_name
+        self._task_name = cfg.task_name
+        if "frame_skip" in cfg:
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:103]: use default_config and deep_merge_dicts to deal with this case

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: setup.py (hunk 10 above)
Comment (zjowowen) [reply setup.py:56]: It seems to be a mistake...

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+            self._frame_skip = cfg.frame_skip
+        else:
+            self._frame_skip = 1
+        if "from_pixels" in cfg:
+            self._from_pixels = cfg.from_pixels
+        else:
+            self._from_pixels = False
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:110]: enable `from_pixels=True` by default

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+        assert cfg.domain_name in dmc2gym_env_info, '{}/{}'.format(cfg.domain_name, dmc2gym_env_info.keys())
+        assert cfg.task_name in dmc2gym_env_info[
+            cfg.domain_name], '{}/{}'.format(cfg.task_name, dmc2gym_env_info[cfg.domain_name].keys())
+
+        self._domain_name = cfg.domain_name
+        self._task_name = cfg.task_name
+        if "frame_skip" in cfg:
+            self._frame_skip = cfg.frame_skip
+        else:
+            self._frame_skip = 1
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:106]: frame_skip=3

File: dizoo/dmc2gym/envs/dmc2gym_env.py [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+        if "from_pixels" in cfg:
+            self._from_pixels = cfg.from_pixels
+        else:
+            self._from_pixels = False
+        if "visualize_reward" in cfg:
+            self._visualize_reward = cfg.visualize_reward
+        else:
+            self._visualize_reward = False
+        if "height" in cfg:
+            self._height = cfg.height
```
Comment (PaParaZz1) [inline dizoo/dmc2gym/envs/dmc2gym_env.py:116]: height=100,width=100

File: dizoo/dmc2gym/envs/dmc2gym_env.py (hunk 11 above)
Comment (zjowowen) [reply dizoo/dmc2gym/envs/dmc2gym_env.py]: Thanks, it's a very good suggestion!

File: dizoo/dmc2gym/envs/dmc2gym_env.py (hunk 12 above)
Comment (zjowowen) [reply dizoo/dmc2gym/envs/dmc2gym_env.py]: fixed.

File: dizoo/dmc2gym/envs/dmc2gym_env.py (hunk 13 above)
Comment (zjowowen) [reply dizoo/dmc2gym/envs/dmc2gym_env.py]: fixed.

File: dizoo/dmc2gym/envs/dmc2gym_env.py (hunk 14 above)
Comment (zjowowen) [reply dizoo/dmc2gym/envs/dmc2gym_env.py]: fixed.

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**